src/pipx/shared_libs.py

```diff
@@ -107,8 +107,6 @@ def upgrade(
                         *_pip_args,
                         "--upgrade",
                         "pip",
```

Comment:
This should set the minimum version of pip that doesn't need setuptools/wheel:
```suggestion
                         "pip >= 23.1",
```